pandas/core/ops.py

```diff
@@ -407,13 +408,36 @@ def _convert_for_datetime(self, lvalues, rvalues):
             else:
                 self.dtype = 'datetime64[ns]'
             mask = isnull(lvalues) | isnull(rvalues)
-            lvalues = lvalues.view(np.int64)
-            rvalues = rvalues.view(np.int64)
+
+            # if adding single offset try vectorized path
+            # in DatetimeIndex; otherwise elementwise apply
+            if self.is_offset_lhs:
+                if len(lvalues) == 1:
```

Comment:
in theory one could provide a series of offsets that are all the same (and equal to each other), but that is prob a special case.